Core api reference refresh (#669)
ROCm/rocDecode: docs/reference/rocDecode-sw-decoder.rst

.. meta::
  :description: The rocDecode software decoder 
  :keywords: decode, video decoder, video decoding, rocDecode, core APIs, AMD, ROCm

********************************************************************

> The rocDecode software decoder API

Reviewer: ```suggestion
rocDecode software decoder API

```

Author: I like The the.

.. code:: cpp

  RocDecoderHostCreateInfo create_info = {};
  create_info.codec_type = rocdec_codec_id;
  create_info.num_decode_threads = 0;     // default
  create_info.max_width = DEFAULT_WIDTH;
  create_info.max_height = DEFAULT_HEIGHT;
  create_info.chroma_format = rocDecVideoChromaFormat_420;
  create_info.output_format = rocDecVideoSurfaceFormat_P016;
  create_info.bit_depth_minus_8 = 2;
  create_info.num_output_surfaces = 1;
  create_info.user_data = &dec_info;
  rocDecCreateDecoderHost(&dec_info.decoder, &create_info);

``rocDecGetDecoderCapsHost()`` queries the capabilities of the underlying software video decoder. Decoder capabilities usually include supported codecs, maximum resolution, and
bit depth.

``rocDecDecodeFrameHost()`` is used to submit frames for software decoding. ``rocDecDecodeFrameHost()`` takes the decoder handle and the pointer to the ``RocdecPicParamsHost`` struct and initiates the video decoding. ``RocdecPicParamsHost`` is populated with the decoded frame information.

> The ``pfn_sequence_callback`` callback is triggered when a format change occurs or when a new sequence header is encountered. This callback needs to be registered in any application that uses the software decoder and its implementation needs to call ``rocDecReconfigureDecoderHost()`` to reconfigure the decoder to handle the new sequence or format. 

Reviewer: ```suggestion
The ``pfn_sequence_callback`` callback is triggered when a format change occurs or when a new sequence header is encountered. This callback must be registered in any application that utilizes the software decoder, and its implementation must call ``rocDecReconfigureDecoderHost()`` to reconfigure the decoder to handle the new sequence or format. 

```

Author: "uses" is a better word than "utilizes".

``rocDecGetDecodeStatusHost()`` can be called to query the decoding status of a frame. The result of the query is either ``rocDecodeStatus_Success``, if decoding is complete, or ``rocDecodeStatus_InProgress``, if decoding is still in progress.

The ``pfn_display_picture`` callback is triggered when a frame has been decoded. This callback needs to be registered by any application that uses the software decoder and its implementation needs to call ``rocDecGetVideoFrameHost()``. ``rocDecGetVideoFrameHost()`` returns the decoded frame's host memory pointer. The decoded frame can then be further processed using this pointer.

> ``rocDecGetVideoFrameHost()`` maps the video ID of the decoded frame to host memory. This is a blocking call that only returns once frame decoding and memory mapping is complete. It returns the host memory pointer as well as information about the :doc:`output surface type <../conceptual/rocDecode-memory-types>`. 

Reviewer: ```suggestion
``rocDecGetVideoFrameHost()`` maps the video ID of the decoded frame to host memory. This is a blocking call that only returns once frame decoding and memory mapping are complete. It returns the host memory pointer as well as information about the :doc:`output surface type <../conceptual/rocDecode-memory-types>`. 

```
